wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/JpaWorkspaceActivityDao.java

```diff
@@ -227,57 +232,84 @@ public WorkspaceActivity findActivity(String workspaceId) throws ServerException
   }
 
   @Override
-  @Transactional(rollbackOn = {ConflictException.class, ServerException.class})
   public void createActivity(WorkspaceActivity activity) throws ConflictException, ServerException {
     try {
-      EntityManager em = managerProvider.get();
-      em.persist(activity);
-      em.flush();
-    } catch (EntityExistsException e) {
+      doCreate(activity);
+    } catch (IntegrityConstraintViolationException e) {
+      throw new ServerException(
+          String.format(
+              "Can not create activity record since the specified workspace with "
+                  + "id '%s' does not exist.",
+              activity.getWorkspaceId()),
+          e);
+    } catch (DuplicateKeyException e) {
       throw new ConflictException(
           "Activity record for workspace ID " + activity.getWorkspaceId() + " already exists.", e);
     } catch (RuntimeException e) {
       throw new ServerException(e.getLocalizedMessage(), e);
```

Comment:
Should we use `.getMessage()` as in the `remove()` or `.getLocalizedMessage()` as is done here?

Reply:
:+1: Will sync them

Reply:
changed to getMessage()